source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+                    binaryForKey.put("base64", keyId);
+                    binaryForKey.put("subType", "04");
+                    JSONObject binWrapper = new JSONObject();
+                    binWrapper.put("$binary", binaryForKey);
+                    
+                    JSONObject encryptedFieldWrap = new JSONObject();
+                    JSONObject encrypt = new JSONObject();
+
+                    encrypt.put("bsonType", chosenBsonType);
+                    if(isDeterministic){
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic");
+                    }else{
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Random");
+                    }
+                    encrypt.put("keyId", new JSONArray().put(binWrapper));
+                    encryptedFieldWrap.put("encrypt", encrypt);
+                    return encryptedFieldWrap;
+                }
+                public static JSONObject getStandardField(String chosenBsonType){
+                    JSONObject standardField = new JSONObject();
+                    standardField.put("bsonType", chosenBsonType);
+                    standardField.put("description", "must be a " + chosenBsonType + " and is required");
+                    return standardField;
+                }
+                public static String getJSONSchema(String keyId){
+                    JSONObject properties = new JSONObject();
+                    properties.put("fullName", getStandardField("string"));
+                    properties.put("address", getEncryptedField(keyId, "string", false));
+                    properties.put("ssn", getEncryptedField(keyId, "int", true));
+                    properties.put("last4SSN", getStandardField("int"));
+                    JSONObject patientInfo = new JSONObject();
+                    patientInfo.put("bsonType", "object");
+                    JSONObject patientInfoProperties = new JSONObject();
+                        patientInfoProperties.put("phone", getEncryptedField(keyId, "int", true));
+                        patientInfoProperties.put("provider", getStandardField("string"));
+                    patientInfo.put("properties", patientInfoProperties);
+                    properties.put("patientInfo", patientInfo);
+
+                    JSONObject medicalRecords = getEncryptedField(keyId, "array", false);
+                    properties.put("medicalRecords",medicalRecords);
+
+                    
+                    JSONObject patientsSchema = new JSONObject();
+                    patientsSchema.put("properties", properties);
+                    patientsSchema.put("bsonType", "object");
+                    String patientsSchemaAsString = patientsSchema.toString();
+                    return patientsSchemaAsString;
+                }
+                public static MongoClient createSecureClient(){
+                    ...
```

Comment:
Question: Should we steer the user to creating the JSON Schema on both the collection (on the server) as well as the client and show them how to accomplish both here? Or do you think that should belong in another section?

Reply:
@nlarew

Reply:
Hmmm not sure tbh. For now let's focus on getting it done client-side and then when the rest of the guide is filled out we can decide where the best home for server-side JSON schema is.

Reply:
Cool

Reply:
💯